src/Facets/EcoFacet.sol

```diff
@@ -200,6 +207,16 @@ contract EcoFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable, LiFiData {
             destination = uint64(_bridgeData.destinationChainId);
         }
 
+        bytes32 intentHash = _getIntentHash(
+            destination,
+            _ecoData.encodedRoute,
+            reward
+        );
+
+        if (PORTAL.getRewardStatus(intentHash) != IEcoPortal.Status.Initial) {
+            revert IntentAlreadyFunded();
```

Comment:
line in tests in not covered. I suggest to copy this PoC code that burrasec testPoc_DuplicateBridgeCallAllowsFundTheft and modify it a bit